src/ko/wolfdotcom/src/eu/kanade/tachiyomi/extension/ko/wolfdotcom/Wolf.kt

```diff
@@ -0,0 +1,398 @@
+                        val type = url.queryParameter("type1")!!
+                        FilterData(
+                            type = type,
+                            typeDisplayName = when (type) {
+                                "day", "complete" -> displayName[0]
+                                "genre" -> displayName[1]
+                                "alphabet" -> displayName[2]
+                                else -> null
+                            },
+                            value = url.queryParameter("type2"),
+                            valueDisplayName = it.ownText(),
+                        )
+                    }
+        } catch (e: Throwable) {
+            Log.e(name, "error parsing filters", e)
+            filterParseError = true
+        }
+    }
+
+    private lateinit var browseCache: List<List<BrowseItem>>
+
+    class BrowseItem(
+        val id: Int,
+        val title: String,
+        val cover: String?,
+    )
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        if (query.isNotBlank()) {
+            return querySearch(query)
+        }
+
+        return if (page == 1) {
+            client.newCall(searchMangaRequest(page, query, filters))
+                .asObservableSuccess()
+                .map {
+                    parseBrowsePage(it)
+                    paginatedBrowsePage(0)
+                }
+        } else {
+            Observable.just(
+                paginatedBrowsePage(page - 1),
+            )
+        }
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/$browsePath".toHttpUrl().newBuilder().apply {
+            filters.filterIsInstance<UrlPartFilter>().forEach { filter ->
+                filter.addToUrl(this)
+            }
+        }.build()
+
+        return GET(url, headers)
+    }
+
+    private fun parseBrowsePage(response: Response) {
+        val document = response.asJsoup()
+
+        parseSearchFilters(document)
+
+        browseCache = document.select(".webtoon-list > ul > li > a").map {
+            val id = it.absUrl("href").toHttpUrl()
+                .queryParameter("toon")!!.toInt()
+
+            BrowseItem(
+                id = id,
+                title = it.selectFirst(".txt > .subject")!!.ownText(),
+                cover = it.selectFirst(".img > img")?.attr("data-original"),
+            )
+        }.chunked(20)
+    }
+
+    private fun paginatedBrowsePage(index: Int): MangasPage {
+        return MangasPage(
+            browseCache[index].map {
+                SManga.create().apply {
+                    url = it.id.toString()
+                    title = it.title
+                    thumbnail_url = it.cover
+                }
+            },
+            browseCache.lastIndex > index,
+        )
+    }
+
+    private val specialChars = Regex("""[^가-힣0-9a-zA-Z ]""", RegexOption.IGNORE_CASE)
```

Comment:
EUC-KR (as used later) supports more than this, however maybe a regex isn't the right approach? I think there are some other Chinese characters, but have not included them below. Regex (x mode):
```regex
[
\p{ASCII}
\p{InHangul_Jamo}
\p{InHangul_Compatibility_Jamo}
\p{InHangul_Syllables}
\p{InHalfwidth_and_Fullwidth_Forms}
]
```

Reply:
I lifted this regex from the site so not sure if including additional characters is good idea

`https://wfwf363.com/assets/js/webtoon.js?v=33` line 146